Runtime/Patterns/Singleton/MonoSingleton.cs

```diff
@@ -57,6 +57,12 @@ public static T Instance
         /// </summary>
         public static void Instantiate()
         {
+            if (HasInstance)
+            {
+                Debug.LogWarning($"You are trying to Instantiate {typeof(T).FullName}, but it already has an Instance. Please use Instance property instead.");
+                return;
+            }
+            
             var name = typeof(T).FullName;
             s_Instance = new GameObject(name).AddComponent<T>();
         }

```

Comment:
I am wondering if we are using 'Instance' now anyways, do you think we still need 'Instance' as public static method?
Or maybe we can just create that instance on demand?


Also would be nice to add test that covers an issue you found, for example we can test that after calling 'Instantiate' multiplayer time we will still have once instance of the 'MonoBehaviourCallback'
What do you think?

Reply:
1. Hided Instantiate() method
2. Added test to validate single Instance of MonoSingleton
3. Incremented semantic version to 2.0.0 because of breaking changes with Instantiate method